aws-lc-rs/src/agreement.rs

```diff
@@ -226,33 +221,29 @@ impl EphemeralPrivateKey {
     pub fn generate(alg: &'static Algorithm, _rng: &dyn SecureRandom) -> Result<Self, Unspecified> {
         match alg.id {
             AlgorithmID::X25519 => {
-                let mut priv_key = [0u8; X25519_PRIVATE_KEY_LEN];
-                let mut pub_key = [0u8; X25519_PUBLIC_VALUE_LEN];
-                unsafe {
-                    X25519_keypair(pub_key.as_mut_ptr(), priv_key.as_mut_ptr());
-                }
+                let priv_key = generate_x25519()?;
                 Ok(EphemeralPrivateKey {
-                    inner_key: KeyInner::X25519(priv_key),
+                    inner_key: KeyInner::X25519(LcPtr::from(priv_key)),
```

Comment:
`generate_x25519` and `ec_key_generate` could return `Result<LcPtr<*mut EVP_PKEY>, Unspecified>` -- I don't see anywhere that we call `.detach()` on the returned values.